java/base-client/src/main/java/no/bankaxept/epayment/client/base/BaseClient.java

```diff
@@ -51,6 +52,16 @@ public static BaseClient withSuppliedToken(String baseurl, Supplier<String> toke
         return new BaseClient(baseurl, new SuppliedAccessTokenPublisher(tokenSupplier));
     }
 
+    private Map<String, List<String>> filterHeaders(Map<String, List<String>> headers, String correlationId, boolean hasBody) {
+        var filteredHeaders = new LinkedHashMap<>(headers);
+        filteredHeaders.put("X-Correlation-Id", List.of(correlationId));
+        if (!(tokenPublisher instanceof EmptyAccessTokenPublisher))
+            filteredHeaders.put("Authorization", List.of("Bearer " + new AccessTokenSubscriber(tokenPublisher).get(tokenTimeout)));
+        if (hasBody && !headers.containsKey("Content-Type"))
+            filteredHeaders.put("Content-Type", List.of("application/json"));
+        return filteredHeaders;
+    }
```

Comment:
Should APIM subscription key be included here?

Reply:
isn't that just used to fetch tokens? no?

Reply:
I think it is used with all APIM requests.

The plan was to use these keys to fetch tokens, and then use the tokens to validated other APIM requests. But we did not get that far, and now we might go another way.

Maybe @thxmasj can correct me on this!

Reply:
What is an apim request?

Reply:
A request that goes through APIM instead of directly to the kubernetes service.

Example:
kubernetes service - http://ebax-payment-api-service
APIM - https://t1-api.techcloud0dev.net/bankaxept-epayment/merchant-api

Reply:
It looks like this was actually needed to make stuff work on T1, huh

Reply:
👍